### ISSUE TITLE
is it possible to optimise camera FOV?

### ISSUE THREAD

**Issue by fairydora** [facebookresearch/pytorch3d#619]
## ❓ Questions on how to use PyTorch3D

Hello :)
Is it possible to optimise camera FOV? I haven't been successful. 

I'm optimising successfully euler angles and camera position.

I've declared all these parameters as nn.Parameters in my model.
self.camera_position = nn.Parameter( torch.from_numpy(np.array(start_cam_T, dtype=np.float32)).to(self.device))
self.camera_rotation = nn.Parameter( torch.from_numpy(np.array(start_cam_R, dtype=np.float32)).to(self.device))
self.fov = nn.Parameter(torch.tensor(fov, requires_grad=True, device=self.device))

In the forward method I do the following:
R = pytorch3d.transforms.euler_angles_to_matrix(self.eulers, convention="XYZ")
T = self.camera_position[None, :]

image = self.renderer(meshes_world=self.meshes, R=R, T=T, cameras=camera, fov=self.fov)

While R and T optimisation works perfectly, fov is not getting optimised. 
Thank you for any advice :)

**Comment by nikhilaravi** [comment]
@fairydora are you trying to modify the camera position optimization tutorial? 

> fov is not getting optimised.

Have you inspected the gradients? Are they zero for fov?

**Comment by fairydora** [comment]
@nikhilaravi     Yes, I did check the grads, and they were ok for the position and euler angles which I optimise but not for fov, which was None - this is what prompted the question. 
So what I think has happened is, it seems that T and R get updated in the renderer for the camera, but fov doesn't.
So I need to update it manually for everything to work :) 
All good, sorry to waste your time! have a nice evening.
